### PR TITLE
[DO NOT MERGE] Search SDK: Fix for issue :3309 SearchSDK tries to read a property marked with JsonIgnore.

### PR DESCRIPTION
**Pull request by updixit**
SearchSDK tries to read a property marked with JsonIgnore.

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by brjohnstmsft** [comment]
@updixit In future pull requests, please make your changes on a topic branch in your fork instead of in the `vs17Dev` branch as this will save you from merge issues later on.

@shahabhijeet We'll wait for the dust to settle on your JSON.NET 10 fix before merging this and bumping the NuGet package and assembly versions. I assume this will happen sometime after the `psSdkJson6` branch is retired. Please let me know if this is not correct. Thanks!

**Comment by brjohnstmsft** [comment]
@idg10 Just an FYI in case you're curious; No action needed on your part. :)

**Review by bitsforthought — APPROVED** [review]

**Comment by shahabhijeet** [comment]
@updixit @brjohnstmsft can we add a test that will catch newtonsoft error that is reported in this issue
https://github.com/Azure/azure-sdk-for-net/issues/2552

**Comment by brjohnstmsft** [comment]
@shahabhijeet I was planning to do that as a separate PR. Would that be ok?

**Comment by shahabhijeet** [comment]
@updixit just noticed, this is targeted towards vs17Dev branch, current branch is psSfkJson6 branch.
can you retarget, then this is good to go.
you can retarget by editing title

**Comment by brjohnstmsft** [comment]
@shahabhijeet is `psSdkJson6` going to be the new default branch permanently? I thought the plan was for it to be merged back into `vs17Dev` at some point. We're just waiting for that to finish so we can release a few changes all at once.

**Comment by shahabhijeet** [comment]
@updixit @brjohnstmsft we have been working on merging the branch back to vs17Dev, but unless PowerShell is updated with the required changes, this merge cannot happen.
So unless you hear back from us regarding this merge, psSdkJson6 is the development branch.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
I don't see nuget version bump.
Is the plan is to publish this change, if yes then the version bump is essential.

**Comment by brjohnstmsft** [comment]
@shahabhijeet This is only one of a few changes we want to include in the next version. I'll follow up with you offline to co-ordinate the roll out of these changes.

**Comment by shahabhijeet** [comment]
@updixit please reopen this PR once it's ready to be merged. This has still CI failing.

**Comment by brjohnstmsft** [comment]
These changes were merged via a different PR. This one should never be re-opened.

**Pull request closed without merging**
